pkg/network/namescheme/networknamescheme_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package namescheme_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+)
+
+var _ = Describe("Network Name Scheme", func() {
+	Context("CreateNetworkNameScheme", func() {
+		DescribeTable("should return the expected NetworkNameSchemeMap",
+			func(networkList []virtv1.Network, expectedNetworkNameSchemeMap map[string]string) {
+				podIfacesNameScheme := namescheme.CreateNetworkNameScheme(networkList)
+
+				Expect(podIfacesNameScheme).To(Equal(expectedNetworkNameSchemeMap))
+			},
+			Entry("when network list is nil", nil, map[string]string{}),
+			Entry("when no multus networks exist",
+				[]virtv1.Network{
+					newPodNetwork("default"),
+				},
+				map[string]string{
+					"default": namescheme.PrimaryPodInterfaceName,
+				}),
+			Entry("when default multus networks exist",
```

Comment:
would it make sense to test a scenario with a multus default network + pod network ? 

Does our admitter reject that ?

Reply:
that's a non-valid configuration. it should error, but not by the networknameshceme

Reply:
Please just make sure the admitter rejects it - I really think it should.

Just saying that unless the admitter blocks it, it may reach here.

Reply:
checking.. will update on this thread

Reply:
Indeed the admitter is [blocking](https://github.com/kubevirt/kubevirt/blob/5f4d9957608c4ae3d8fd1fe40a9a70f58da50408/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L176-L178) the config you mention